### ISSUE TITLE
Allow to include regularization terms in the loss function

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gate-lf-pytorch-json#22]
This means the loss function depends, like the optimizer, on potentially all model parameters,
so we have to construct it at the same point. 

Initially just make it work like this:
* L2 norm of parms, weighted by w1
* L1 norm of parms, weighted by w2
* w0 for original loss criterion
* w1, w2, w0 re-scaled to add to 1.0
